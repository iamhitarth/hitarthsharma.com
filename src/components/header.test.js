import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Header from './header'

describe('Header', () => {
  const siteTitle = 'Hitarth Sharma'

  afterEach(cleanup)

  it('matches snapshot', () => {
    const { container } = render(<Header siteTitle={siteTitle} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('has site title', () => {
    const { getByText } = render(<Header siteTitle={siteTitle} />)
    expect(getByText(siteTitle)).not.toBeNull()
  })

  it('has nav links', () => {
    const { getByText } = render(<Header siteTitle={siteTitle} />)
    expect(getByText('Blog')).not.toBeNull()
    expect(getByText('About')).not.toBeNull()
  })
})
