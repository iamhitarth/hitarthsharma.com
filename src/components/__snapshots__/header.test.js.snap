// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div
  class="sc-EHOje fWrQAX"
>
  <div
    class="sc-ifAKCX CToAk"
  >
    <h1
      class="sc-bdVaJa bcSwbK"
    >
      <link
        class="sc-bwzfXH gRlGoj"
        to="/"
      >
        Hitarth Sharma
      </link>
    </h1>
    <ul
      class="sc-htpNat bGvYfA"
    >
      <li
        class="sc-bxivhb lmyeQp"
      >
        <link
          class="sc-bwzfXH gRlGoj"
          to="/"
        >
          Home
        </link>
      </li>
      <li
        class="sc-bxivhb lmyeQp"
      >
        <link
          class="sc-bwzfXH gRlGoj"
          to="/about"
        >
          About
        </link>
      </li>
    </ul>
  </div>
</div>
`;
